modules/javafx.graphics/src/test/java/test/javafx/scene/paint/StopListTest.java

```diff
@@ -151,4 +152,134 @@ public void testNormalizeStopsNonUnsorted() {
+            var secondList = normalize(List.of(new Stop(0, color1), new Stop(1, color3)));
+
+            var expected = List.of(
+                new Stop(0, color1),
+                new Stop(0.5, color2.interpolate(color3, 0.5)),
+                new Stop(1, color2));
+            assertEquals(expected, StopShim.interpolateLists(firstList, secondList, 0.5));
+
+            // An interpolation factor close to zero should yield a stop list very similar to the first list
+            expected = firstList;
+            assertSimilar(expected, StopShim.interpolateLists(firstList, secondList, 0.001));
+
+            // An interpolation factor close to one should yield a stop list very similar to the second list,
+            // but with a different number of stops (3 instead of 2).
+            expected = List.of(new Stop(0, color1), new Stop(0.5, color2), new Stop(1, color3));
+            assertSimilar(expected, StopShim.interpolateLists(firstList, secondList, 0.999));
+        }
+
+        @Test
+        public void firstListSmallerThanSecondList_sameFirstAndLastOffset() {
+            var firstList = normalize(List.of(new Stop(0, color1), new Stop(1, color3)));
+            var secondList = normalize(List.of(new Stop(0, color1), new Stop(0.5, color3), new Stop(1, color1)));
+
+            var expected = List.of(
+                new Stop(0, color1),
+                new Stop(0.5, color2.interpolate(color3, 0.5)),
+                new Stop(1, color2));
+            assertEquals(expected, StopShim.interpolateLists(firstList, secondList, 0.5));
+
+            // An interpolation factor close to zero should yield a stop list very similar to the first list,
+            // but with a different number of stops (3 instead of 2).
+            expected = List.of( new Stop(0, color1), new Stop(0.5, color2), new Stop(1, color3));
+            assertSimilar(expected, StopShim.interpolateLists(firstList, secondList, 0.001));
+
+            // An interpolation factor close to one should yield a stop list very similar to the second list.
+            expected = secondList;
+            assertSimilar(expected, StopShim.interpolateLists(firstList, secondList, 0.999));
+        }
+
+        @Test
+        public void differentSize_differentOffsets() {
+            double t = 0.5;
+            var firstList = normalize(List.of(new Stop(0, color1), new Stop(1, color3)));
+            var secondList = normalize(List.of(new Stop(0.25, color3), new Stop(0.75, color1)));
+            var res = StopShim.interpolateLists(firstList, secondList, t);
+            assertEquals(List.of(
+                new Stop(0, color2),
+                new Stop(0.25, color1.interpolate(color3, 0.25).interpolate(color3, t)),
+                new Stop(0.75, color1.interpolate(color3, 0.75).interpolate(color1, t)),
+                new Stop(1, color2)),
+            res);
+        }
```

Comment:
Do you think any other combinations should be added, for completeness sake?

For example, let's consider two lists with 2 points each.  We might want to test scenarios:

```
--
     --

--|
  |--

---
 ---

----|
 ---|

------
  --

  |-----
  |--

  |---|
  |---|

    |--
  --|

     ---
--
```

and also have the two lists swapped maybe?

what do you think?

Reply:
I've added another test, which gives us 100% coverage for all code paths. I would consider this complete.